test/unit/classification_test.rb

```diff
@@ -302,6 +302,26 @@ def setup
                  classification.send(:values_hash))
   end
 
+  test 'smart class parameter with use_puppet_default on specific matcher does not send a value to puppet' do
+    key = FactoryGirl.create(:puppetclass_lookup_key, :as_smart_class_param,
+                             :override => true, :key_type => 'string',
+                             :default_value => "123", :path => "organization\nos\nlocation",
+                             :puppetclass => puppetclasses(:one))
+
+    as_admin do
+      LookupValue.create! :lookup_key_id => key.id,
+                          :match => "location=#{taxonomies(:location1)}",
+                          :value => "345",
+                          :use_puppet_default => true
+    end
+
+    key.reload
+
+    enc = classification.enc
+    assert enc['base'][key.key].nil?
```

Comment:
I'm unsure if this check is sufficient, partly because of limited ruby knowledge. I think the key should not be present at all. In python I'd write ```assert key.key not in enc['base']```.